### PR TITLE
Preparations for fetching TVL directly from FE2

### PR DESCRIPTION
**Pull request by imxeno**

### CODE REVIEW

File: packages/frontend2/src/app/(new)/(other)/scaling-next/summary/page.tsx [hunk 1]
```diff
@@ -20,11 +21,44 @@ export const metadata = getDefaultMetadata({
...
+        number,
+        number,
+        number,
+        number,
+        number,
+        number,
+        number,
+        number,
+      ],
+  )
```
Comment (sdlyy) [inline packages/frontend2/src/app/(new)/(other)/scaling-next/summary/page.tsx:47]: Extract to separate function or/and add comments around this transformation - ik why it is there, somebody else might not

File: packages/frontend2/src/app/api/dev/route.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+export const dynamic = 'force-dynamic'
+
+export async function GET() {
+  const chart = await getTvlChart({ range: '7d', type: 'layer2' })
+  return NextResponse.json({
+    length: chart.length,
+    chart: chart.map((c) => [
+      new Date(c[0] * 1000).toISOString(),
+      `Sum: ${c[1] + c[2] + c[3]}`,
+    ]),
```
Comment (sdlyy) [inline packages/frontend2/src/app/api/dev/route.ts:13]: Some debug val leftover?

File: packages/frontend2/src/app/(new)/(other)/scaling-next/summary/page.tsx [hunk 3]
```diff
@@ -20,11 +21,44 @@ export const metadata = getDefaultMetadata({
...
+      <TvlChart
+        data={{
+          ...tvl.layers2s,
+          hourly: {
+            ...tvl.layers2s.hourly,
+            data: hourly,
+          },
+        }}
+        milestones={HOMEPAGE_MILESTONES}
+      />
```
Comment (sdlyy) [inline packages/frontend2/src/app/(new)/(other)/scaling-next/summary/page.tsx:61]: Same as above, why do we need to construct such an override explicitly?

File: packages/frontend2/src/server/features/tvl/get-tvl-chart.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  return Object.entries(timestampValues).map(([timestamp, values]) => {
+    const summed = sumValuesPerSource(values, true)
+    return [
+      +timestamp,
+      Number(summed.native),
+      Number(summed.canonical),
+      Number(summed.external),
+      ethPrices[+timestamp]!,
+    ] as const
+  })
```
Comment (sdlyy) [inline packages/frontend2/src/server/features/tvl/get-tvl-chart.ts:55]: Do we use `false`/`undefined` flag in this function elsewhere?

File: packages/frontend2/src/server/features/tvl/get-tvl-projects.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  ]
+
+  const tvlAmounts = getTvlAmountsConfig(projects)
+
+  return projects.flatMap(({ projectId, type, slug }) => {
+    const amounts = tvlAmounts.filter((o) => o.project === projectId)
+    if (!amounts) {
+      return []
+    }
+    assert(amounts, 'Config not found: ' + projectId.toString())
```
Comment (sdlyy) [inline packages/frontend2/src/server/features/tvl/get-tvl-projects.ts:41]: Will this case fall through? We already have truthy check above

File: packages/frontend2/src/server/features/tvl/get-tvl-target-timestamp.ts
Comment (sdlyy) [inline packages/frontend2/src/server/features/tvl/get-tvl-target-timestamp.ts:1]: I presume we have to keep it in sync with backend tvl configuration, should we add a comment here or reference?

File: packages/frontend2/src/server/features/tvl/get-tvl-values-for-projects.ts [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    ).map((i) => {
+      return minTimestamp.add(
+        i * (resolution === 'sixHourly' ? 6 : 1),
+        resolution === 'hourly'
+          ? 'hours'
+          : resolution === 'sixHourly'
+            ? 'hours'
+            : 'days',
+      )
+    })
```
Comment (sdlyy) [inline packages/frontend2/src/server/features/tvl/get-tvl-values-for-projects.ts:67]: Could we extract it into a separate function?

File: packages/frontend2/src/server/features/tvl/get-tvl-values-for-projects.ts [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+    for (const timestamp of timestamps) {
+      const values = (valuesByTimestamp[timestamp.toString()] ?? []).filter(
+        (v) => {
+          if (status.excluded.has(v.dataSource)) {
+            return false
+          }
+          const projectSource = project.sources.get(v.dataSource)
+          if (!projectSource) return false
```
Comment (sdlyy) [inline packages/frontend2/src/server/features/tvl/get-tvl-values-for-projects.ts:76]: Mixed if-context style, unify please

**Review by sdlyy — COMMENTED** [review]
rough pass, I'll the the second one later on

File: packages/frontend2/src/app/api/dev/route.ts (hunk 2 above)
Comment (imxeno) [reply packages/frontend2/src/app/api/dev/route.ts:13]: Yeah, this route should be removed.

File: packages/frontend2/src/app/(new)/(other)/scaling-next/summary/page.tsx (hunk 3 above)
Comment (imxeno) [reply packages/frontend2/src/app/(new)/(other)/scaling-next/summary/page.tsx:61]: This and the above cast is only left for dev purposes. To not add any more complexity I'll do it as part of L2B-6115.

File: packages/frontend2/src/server/features/tvl/get-tvl-chart.ts (hunk 4 above)
Comment (imxeno) [reply packages/frontend2/src/server/features/tvl/get-tvl-chart.ts:55]: For now - no. Later we'll use it when summing values that are not coming from different projects (like project TVL chart).

File: packages/frontend2/src/server/features/tvl/get-tvl-projects.ts (hunk 5 above)
Comment (imxeno) [reply packages/frontend2/src/server/features/tvl/get-tvl-projects.ts:41]: Leftover from the old code :) Removing!

File: packages/frontend2/src/app/(new)/(other)/scaling-next/summary/page.tsx (hunk 3 above)
Comment (imxeno) [reply packages/frontend2/src/app/(new)/(other)/scaling-next/summary/page.tsx]: Anyway - I reverted those changes here.

File: packages/frontend2/src/server/features/tvl/get-tvl-target-timestamp.ts
Comment (imxeno) [reply packages/frontend2/src/server/features/tvl/get-tvl-target-timestamp.ts:1]: Actually not really - we just need to account for the "safe" timestamp which is `currentHour - 1`. I do `-2` to keep current behavior.

**Review by sdlyy — APPROVED** [review]

**Pull request merged**